component/loki/source/file/metrics.go

```diff
@@ -0,0 +1,61 @@
+package file
+
+import "github.com/prometheus/client_golang/prometheus"
+
+// metrics hold the set of file-based metrics.
+type metrics struct {
+	// Registerer used. May be nil.
+	reg prometheus.Registerer
+
+	// File-specific metrics
+	readBytes        *prometheus.GaugeVec
+	totalBytes       *prometheus.GaugeVec
+	readLines        *prometheus.CounterVec
+	encodingFailures *prometheus.CounterVec
+	filesActive      prometheus.Gauge
+}
+
+// newMetrics creates a new set of file metrics. If reg is non-nil, the metrics
+// will be registered.
+func newMetrics(reg prometheus.Registerer) *metrics {
+	var m metrics
+	m.reg = reg
+
+	m.readBytes = prometheus.NewGaugeVec(prometheus.GaugeOpts{
```

Comment:
nit: Don't use the Namespace field when creating metrics because it makes it harder to find where in the code metrics are coming from. I'd prefer us only using the name field, like `Name: "loki_source_file_read_bytes_total"`.